Add release notes of 5.6.0 (#2398)
emqtt/emq-docs-cn: en_US/changes/changes-ce-v5.md

  - **Metrics expansion with cluster sessions gauge**: A new gauge metric, `cluster_sessions`, is added to better track the number of sessions within the cluster. This metric is also integrated into Prometheus for easy monitoring:

    ```
    # TYPE emqx_cluster_sessions_count gauge
    emqx_cluster_sessions_count 1234
    ```

    NOTE: Please consider this metric as an approximate estimation. Due to the asynchronous nature of data collection and calculation, exact precision may vary.

- [#12338](https://github.com/emqx/emqx/pull/12338) Introduced a time-based garbage collection mechanism to the RocksDB-based persistent session backend. This feature ensures more efficient management of stored messages, optimizing storage utilization and system performance by automatically purging outdated messages.

- [#12398](https://github.com/emqx/emqx/pull/12398) Exposed the `swagger_support` option in the Dashboard configuration, allowing for the enabling or disabling of the Swagger API documentation.

> - [#12467](https://github.com/emqx/emqx/pull/12467) Supported cluster discovery using AAAA DNS record type.

Reviewer: ```suggestion
- [#12467](https://github.com/emqx/emqx/pull/12467) Started supporting cluster discovery using AAAA DNS record type.

```

  - `GET /clients/{clientid}/inflight_messages?limit=100&position={position}`

  Where `{position}` is a value (opaque string token) of `meta.position` field from the previous response.

  Ordering and Prioritization:

  - **Mqueue Messages**: These are prioritized and sequenced based on their queue order (FIFO), from higher to lower priority. By default, mqueue messages carry a uniform priority level of 0.
  - **In-Flight Messages**: Sequenced by the timestamp of their insertion into the in-flight storage, from oldest to newest.

- [#12590](https://github.com/emqx/emqx/pull/12590) Removed `mfa` meta data from log messages to improve clarity.

- [#12641](https://github.com/emqx/emqx/pull/12641) Improved text log formatter fields order. The new fields order is as follows:

  `tag` > `clientid` > `msg` > `peername` > `username` > `topic` > [other fields]

- [#12670](https://github.com/emqx/emqx/pull/12670) Added field `shared_subscriptions` to endpoint `/monitor_current` and `/monitor_current/nodes/:node`.

- [#12679](https://github.com/emqx/emqx/pull/12679) Upgraded docker image base from Debian 11 to Debian 12.

> - [#12700](https://github.com/emqx/emqx/pull/12700) Supported "b" and "B" unit in bytesize hocon fields.

Reviewer: ```suggestion
- [#12700](https://github.com/emqx/emqx/pull/12700) Started supporting "b" and "B" unit in bytesize hocon fields.

```

- [#12715](https://github.com/emqx/emqx/pull/12715) Fixed a crash that could occur during configuration updates if the connector for the ingress data integration source had active channels.

- [#12740](https://github.com/emqx/emqx/pull/12740) Fixed an issue when durable sessions could not be kicked out.

- [#12768](https://github.com/emqx/emqx/pull/12768) Addressed a startup failure issue in EMQX version 5.4.0 and later, particularly noted during rolling upgrades from versions before 5.4.0. The issue was related to the initialization of the routing schema when both v1 and v2 routing tables were empty.

  The node now attempts to retrieve the routing schema version in use across the cluster instead of using the v2 routing table by default when local routing tables are found empty at startup. This approach mitigates potential conflicts and reduces the chances of diverging routing storage schemas among cluster nodes, especially in a mixed-version cluster scenario. 

>   If conflict is detected in a running cluster, EMQX writes instructions on how to manually resolve it in the log as part of the error message with `critical` severity. The same error message and instructions will also be written in the `error` level log to make sure this message will not get lost even if no log handler is configured.

Reviewer: > in the `error` level log

this is incorrect, "standard error" is not `error` level log, it is standard linux data stream that can be used by programs to write error messages. See, for example, https://www.howtogeek.com/435903/what-are-stdin-stdout-and-stderr-on-linux/.

Standard error data stream is always available to programs, that's why we write error message and instructions there. Using `error` level log would make no sense, since `critical` is of higher severity, and each error message with `critical` severity will end up in `error` level log anyway. We try to handle situation when user disabled all standard log handlers in the application. In this case they will see the message anyway.

Reviewer 2: Sorry for the mistakes and thank you for the clear explanation. I have corrected the descriptions.

  Examples of Multi-Client/Username Queries:

  - To query multiple clients by ID: `/clients?clientid=client1&clientid=client2`
  - To query multiple users: `/clients?username=user11&username=user2`
  - To combine multiple client IDs and usernames in one query: `/clients?clientid=client1&clientid=client2&username=user1&username=user2`

  Examples of Selecting Fields for the Response:

  - To include all fields in the response: `/clients?fields=all` (Note: Omitting the `fields` parameter defaults to returning all fields.)
  - To specify only certain fields: `/clients?fields=clientid,username`

- [#12381](https://github.com/emqx/emqx/pull/12381) Added new SQL functions: `map_keys()`, `map_values()`, `map_to_entries()`, `join_to_string()`, `join_to_string()`, `join_to_sql_values_string()`, `is_null_var()`, `is_not_null_var()`.

  For more information on the functions and their usage, refer to [Built-in SQL Functions](../data-integration/rule-sql-builtin-functions) the documentation.

> - [#12336](https://github.com/emqx/emqx/pull/12336) Refined the approach to managing asynchronous tasks by segregating the cleanup of channels into its own dedicated pool. This separation addresses performance issues encountered during channels cleanup under conditions of high network latency, ensuring that such tasks do not impede the efficiency of other asynchronous operations, such as route cleanup.

Reviewer 3: ```suggestion
- [#12336](https://github.com/emqx/emqx/pull/12336) Performance enhancement. Created a dedicated async task handler pool to handle client session cleanup tasks.

```

Author: @Meggielqk please help to update Chinese changelog
